src/backend_model.cc

```diff
@@ -162,6 +163,8 @@ TritonModel::Create(
         ValidateInstanceGroup(model_config, min_compute_capability));
   }
 
+  DeviceMemoryTracker::ScopedMemoryUsage usage;
+  DeviceMemoryTracker::TrackThreadMemoryUsage(&usage);
```

Comment:
Do you think we can make the memory usage tracking a self-reported metric by the backend using the backend API instead of automatically tracking it in the core? This would ensure that the value reported can be customized if needed. For example, the Python backend can report the value for the stub process.

Reply:
I was thinking about that but I think this will be an alternative if the backend does provide the usage info. The current way require no changes in the backend implementation so all backends get it for free if they are allocating / freeing memory on demand. We can later on add new field to "backend attribute" API so if the backend is implemented to provide more accurate usage, we will collect through the backend instead of through cupti

Reply:
> The current way require no changes in the backend implementation so all backends get it for free

I think that is great but I think reporting an inaccurate number for Python/FT backends would be bad UX. Maybe we can make the memory tracking a backend utility that the backend can opt-in?

Reply:
That is probably a better idea, but I need to do some experiments on how the CUPIT API usage may need to be changed if moved to backends (that shouldn't need to consider each other). At the minimum, will be backend provided callback to provide memory usage (API for backend to prepare the struct) when requested by core. And depending on CUPIT usage, Triton wrapped API like the memory pool.